EIPS/eip-3675.md

```diff
@@ -0,0 +1,247 @@
+
+To understand the motivation of introducing the Proof-of-Stake consensus see the Motivation section of [EIP-2982](./eip-2982.md#motivation).
+
+
+## Specification
+
+### Definitions
+
+* **PoW block**: Block that is built and verified by the existing proof-of-work mechanism. In other words, a block of the Ethereum network before the consensus upgrade.
+* **PoS block**: Block that is built and verified by the new proof-of-stake mechanism.
+* **Terminal PoW block**: A PoW block that satisfies the following conditions -- 
+`pow_block.total_difficulty >= TERMINAL_TOTAL_DIFFICULTY` *and* `pow_block.parent_block.total_difficulty < TERMINAL_TOTAL_DIFFICULTY`.  
+There can be more than one terminal PoW block in the network, e.g. multiple children of the same pre-terminal block.
+* **`TERMINAL_TOTAL_DIFFICULTY`** The amount of total difficulty reached by the network that triggers the consensus upgrade.
+* **`TRANSITION_BLOCK`** The earliest PoS block of the canonical chain, i.e. the PoS block with the lowest block height.  
+`TRANSITION_BLOCK` **MUST** be a child of a terminal PoW block.
+* **`POS_CONSENSUS_VALIDATED`** An event occurring when the PoS block is validated with respect to the proof-of-stake consensus rules.
+* **`POS_CHAINHEAD_SET`** An event occurring when the PoS block is designated as the head of the canonical chain.
+* **`POS_BLOCK_FINALIZED`** An event occurring when the PoS block is considered as finalized.
+
+#### PoS events
+
+Events having the `POS_` prefix in the name (PoS events) are emitted by the new proof-of-stake consensus mechanism. They signify the corresponding assertion that has been made regarding a block specified by the event. The underlying logic of PoS events can be found in the beacon chain specification. On the occurrence of each PoS event the corresponding action that is specified by this EIP **MUST** be taken.
+
+Each `POS_BLOCK_FINALIZED` event has a reference to the head of the canonical chain in addition to the finalized block and incorporates the semantics of the `POS_CHAINHEAD_SET` event. Thus, every action and effect that is specified on the occurrence of `POS_CHAINHEAD_SET` event according to this EIP **MUST** also be taken on every occurrence of `POS_BLOCK_FINALIZED` event.
+
+
+### PoW block processing
+
+PoW blocks that are descendants of any terminal PoW block **MUST NOT** be imported. This implies that a terminal PoW block will be the last PoW block in the canonical chain.
+
+
+### Block structure
+
+Beginning with `TRANSITION_BLOCK`, a number of previously dynamic block fields are deprecated by enforcing these values to instead be constants. Each block field listed in the table below **MUST** be replaced with the corresponding constant value.
+
+| Field | Constant value |
+|-|-|
+| **`ommersHash`** | `Keccak256(RLP(EMPTY_LIST))`  |
+| **`difficulty`** | `0` |
+| **`extraData`** | `0x80` |
+| **`mixHash`** | `0x0000000000000000000000000000000000000000000000000000000000000000` |
```

Comment:
Since now `extraData` shows the raw RLP value of 0x80, shouldn't all these fields do the same?

Reply:
IMO, this table should rather contain raw values, not their encoding. I am thinking about replacing `0x80` with `EMPTY_BYTE_ARRAY` or `EMPTY_BYTE_STRING`.

Reply:
Maybe have multiple columns here, one that shows the post-serialization value and others that shows how to get that pre-serialization?  Different implementations may have an easier or harder time hard-coding things at different layers of the serialization process, and this ensures that everyone ends up with the same result no matter how they get there.
```
**`ommersHash`** | `0x1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347` | `keccak256([0xc0])` | `keccak256(rlp([]))
**`ommers`** | `0xc0` | `rlp([])`
```

Reply:
This table should look better now, please check.